src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -656,11 +653,12 @@ std::string LocalResourceManager::SerializedTaskResourceInstances(
   return buffer.str();
 }
 
-void LocalResourceManager::UpdateLastResourceUsage(
-    std::shared_ptr<SchedulingResources> gcs_resources) {
+void LocalResourceManager::ResetResources(
+    std::shared_ptr<SchedulingResources> replacement) {
   last_report_resources_ = std::make_unique<NodeResources>(ResourceMapToNodeResources(
-      resource_name_to_id_, gcs_resources->GetTotalResources().GetResourceMap(),
-      gcs_resources->GetAvailableResources().GetResourceMap()));
+      resource_name_to_id_, replacement->GetTotalResources().GetResourceMap(),
+      replacement->GetAvailableResources().GetResourceMap()));
+  OnResourceChanged();
```

Comment:
We didn't actually change local resource, we only change the the one we report to gcs

Reply:
oh you are right!... was a bit confused

Reply:
We shouldn't need to call `OnResourceChanged()` here right since local resource is not changed